bartop-api/test/integration/catalogsEndpointTest.js

```diff
@@ -0,0 +1,140 @@
+    const response = await dbAdapter.r.table('users').insert(users.testUser);
+    userId = response.generated_keys[0];
+    return;
+  });
+
+  describe('Rest', function() {
+    it('POST - create a new catalog', function(done) {
+      request(app)
+        .post('/api/v1/catalogs')
+        .set('Content-Type', 'application/json')
+        .set('Authorization', `Bearer ${token}`)
+        .send({ userId, drinkIds })
+        .end((err, res) => {
+          expect(res.statusCode).to.equal(201);
+          expect(res.body).to.be.an('object');
+          expect(res.body.id).to.equal(userId);
+          expect(res.body.auth0Id).to.equal(users.testUser.auth0Id);
+          expect(res.body.catalog).to.deep.equal(drinkIds);
+          done();
+        });
+    });
+
+    it('POST - throw error if user does not exist', function(done) {
+      const thisError = errors.NOT_FOUND;
+      request(app)
+        .post('/api/v1/catalogs')
+        .set('Content-Type', 'application/json')
+        .set('Authorization', `Bearer ${token}`)
+        .send({ userId: 'Arnald', drinkIds })
+        .end((err, res) => {
+          expect(res.statusCode).to.equal(thisError.code);
+          expect(res.body).to.equal(thisError.message);
+          done();
+        });
+    });
+
+    it('POST - throw error if body does not match model', function(done) {
+      const thisError = errors.BODY_MODEL;
+      request(app)
+        .post('/api/v1/catalogs')
+        .set('Content-Type', 'application/json')
+        .set('Authorization', `Bearer ${token}`)
+        .send({ id: users.postUser.auth0Id })
+        .end((err, res) => {
+          expect(res.statusCode).to.equal(thisError.code);
+          expect(res.body).to.equal(thisError.message);
+          done();
+        });
+    });
+
+    it('POST - throw error if content-type is unsupported', function(done) {
+      const thisError = errors.CONTENT_TYPE;
+      request(app)
+        .post('/api/v1/catalogs')
+        .set('Content-Type', 'multipart/form-data')
+        .set('Authorization', `Bearer ${token}`)
+        // if content-type is not set to json,
+        // the send() method expects a string
+        .send(`{ id: ${users.postUser.auth0Id} }`)
+        .end((err, res) => {
+          expect(res.statusCode).to.equal(thisError.code);
+          expect(res.body).to.equal(thisError.message);
+          done();
+        });
+    });
+  });
+
+  describe('GraphQL', function() {
+    it('Mutation - create a new user', function(done) {
+      const query = `
+        mutation {
+          createCatalog(newCatalog: { userId: "${userId}", drinkIds: ["${drinkIds.join(
+        '", "'
+      )}"] }) {
+            id
+          }
+        }`;
+      request(app)
+        .post('/api/graphql')
+        .set('Authorization', `Bearer ${token}`)
+        .set('Content-Type', 'application/json')
+        .send({ query })
+        .end((err, res) => {
+          const user = res.body.data.createCatalog;
```

Comment:
I think this is some copy pasta peril as well 🚨

Reply:
this is literally every single one of my PRs

Reply:
aha this is actually not a typo. the create catalog function returns a `User` object because that is what is being updated. It's possible to _only_ return the list of drink IDs, do you think that's better? That's almost just getting the request back as a response

Reply:
I for sure didn't finish this test though lol

Reply:
Ah, this is a great article. Thanks for sharing. They are not suggesting that, but they are suggesting creating a specific payload schema type for every mutation. That way I can send whatever I want back and it can change over time without effecting anything else. GitHub does this actually, and I made a mental note to look into why but haven't yet

Reply:
This is actually something I would prefer to resolve in a different PR, since I should go back and update this for `drinks` and `users` too

Reply:
Yee sounds `good`